### PR TITLE
jdk8 MacOS Notarization support

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: andrew-m-leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
Porting of following openjdk jdk8u changes:
- http://cr.openjdk.java.net/~stooke/webrevs/jdk-8226288-jdk8u/01/01/
- http://cr.openjdk.java.net/~stooke/webrevs/jdk-8226288-jdk8u/01/jdk.01/
- http://cr.openjdk.java.net/~dcherepanov/openjdk8u/8235687/webrev.v0/ 
- http://cr.openjdk.java.net/~dcherepanov/openjdk8u/8238225/webrev.v0/

**Comment by andrew-m-leonard** [comment]
jenkins compile xlinux jdk8

**Comment by andrew-m-leonard** [comment]
jenkins compile x86-64_linux jdk8

**Comment by andrew-m-leonard** [comment]
jenkins compile

**Comment by lumpfish** [comment]
Do you need to take the test changes (might existing tests fail when the changes are added)?

File: common/autoconf/libraries.m4 [hunk 1]
```diff
@@ -1047,8 +1047,12 @@ AC_DEFUN_ONCE([LIB_SETUP_STATIC_LINK_LIBSTDCPP],
   fi
 
   # TODO better (platform agnostic) test
-  if test "x$OPENJDK_TARGET_OS" = xmacosx && test "x$LIBCXX" = x && test "x$TOOLCHAIN_TYPE" = xgcc; then
-    LIBCXX="-lstdc++"
+  if test "x$OPENJDK_TARGET_OS" = xmacosx && test "x$LIBCXX" = x ; then
```
Comment (keithc-ca) [inline common/autoconf/libraries.m4:1050]: nit: trailing whitespace

File: common/autoconf/flags.m4 [hunk 2]
```diff
@@ -646,6 +691,22 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_JDK],
       MACOSX_VERSION_MIN=10.9.0
       AC_SUBST(MACOSX_VERSION_MIN)
     
+      # The macro takes the version with no dots, ex: 1070
+      # Let the flags variables get resolved in make for easier override on make
+      # command line.
+      CCXXFLAGS_JDK="$CCXXFLAGS_JDK -DMAC_OS_X_VERSION_MAX_ALLOWED=\$(subst .,,\$(MACOSX_VERSION_MIN)) -mmacosx-version-min=\$(MACOSX_VERSION_MIN)"
+      LDFLAGS_JDK="$LDFLAGS_JDK -mmacosx-version-min=\$(MACOSX_VERSION_MIN)"
+    elif test "x$TOOLCHAIN_TYPE" = xclang; then
```
Comment (keithc-ca) [inline common/autoconf/flags.m4:699]: This branch is the same as for gcc. The test above should be changed:
```
if test "x$TOOLCHAIN_TYPE" = xgcc -o "x$TOOLCHAIN_TYPE" = xclang; then
```

File: common/autoconf/flags.m4 [hunk 3]
```diff
@@ -166,6 +166,26 @@ AC_DEFUN_ONCE([FLAGS_SETUP_COMPILER_FLAGS_FOR_LIBS],
     C_FLAG_REORDER=''
     CXX_FLAG_REORDER=''
 
+    if test "x$OPENJDK_TARGET_OS" = xmacosx; then
```
Comment (keithc-ca) [inline common/autoconf/flags.m4:169]: This file has an IBM copyright notice: the end date should be updated.

Comment (andrew-m-leonard) [reply common/autoconf/flags.m4:169]: even though this change wasn't an IBM change?

**Comment by andrew-m-leonard** [comment]
> Do you need to take the test changes (might existing tests fail when the changes are added)?

I couldn't as those test changes depend on the JFR backport

File: common/autoconf/flags.m4 (hunk 3 above)
Comment (andrew-m-leonard) [reply common/autoconf/flags.m4:169]: it's ok, it now has an IBM change :-)

Comment (keithc-ca) [reply common/autoconf/flags.m4:169]: It had an IBM change before, hence the presence of the copyright notice.

**Comment by keithc-ca** [comment]
@andrew-m-leonard When should we expect an equivalent pull request for the openj9 branch?

**Comment by andrew-m-leonard** [comment]
> @andrew-m-leonard When should we expect an equivalent pull request for the openj9 branch?

@keithc-ca already merged

**Comment by keithc-ca** [comment]
Yes, I saw #407 since I asked. Thanks.

**Pull request merged**